org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/framework/android/AndroidSupport.java

```diff
@@ -0,0 +1,128 @@
+import org.gradle.tooling.model.build.GradleEnvironment;
+
+public class AndroidSupport implements IFrameworkSupport {
+
+	public static String SYNCHRONIZATION_TASKS_API_GRADLE = "5.4";
+	public static String ASSEMBLE_TASK_NAME = "assemble";
+
+	@Override
+	public void onDidProjectsImported(IProgressMonitor monitor) {
+		PreferenceManager preferencesManager = JavaLanguageServerPlugin.getPreferencesManager();
+		if (preferencesManager == null) {
+			return;
+		}
+
+		if (!preferencesManager.getPreferences().isAndroidSupportEnabled()) {
+			return;
+		}
+
+		List<IProject> projects = ProjectUtils.getGradleProjects();
+		if (projects.isEmpty()) {
+			return;
+		}
+
+		JavaLanguageClient client = JavaLanguageServerPlugin.getProjectsManager().getConnection();
+		ProgressReport progressReport = new ProgressReport(UUID.randomUUID().toString());
+		progressReport.setTask("Running Gradle tasks");
+		progressReport.setComplete(false);
+		progressReport.setTotalWork(projects.size());
+		progressReport.setStatus("Generating Android project sources files...");
+		client.sendProgressReport(progressReport);
+
+		// Avoid double-processing a GradleBuild
+		Set<GradleBuild> processedGradleBuilds = new HashSet<>();
+		for (IProject project : projects) {
+			if (!ProjectUtils.isJavaProject(project)) {
+				continue;
+			}
+			Optional<GradleBuild> gradleBuild = GradleCore.getWorkspace().getBuild(project);
+			if (gradleBuild.isEmpty()) {
+				continue;
+			}
+			GradleBuild build = gradleBuild.get();
+			if (processedGradleBuilds.contains(build)) {
+				continue;
+			}
+			processedGradleBuilds.add(build);
+			try {
+				BuildEnvironment buildEnvironment = build.withConnection(connection -> connection.getModel(BuildEnvironment.class), monitor);
+				if (buildEnvironment == null) {
+					continue;
+				}
+				GradleEnvironment gradleEnvironment = buildEnvironment.getGradle();
+				if (gradleEnvironment == null) {
+					continue;
+				}
+				String gradleVersion = gradleEnvironment.getGradleVersion();
+				if (gradleVersion == null) {
+					continue;
+				}
+				// for Gradle versions support eclipseModel.synchronizationTasks() API, there is nothing to do after importing
+				if (GradleVersion.version(gradleVersion).compareTo(GradleVersion.version(SYNCHRONIZATION_TASKS_API_GRADLE)) >= 0) {
+					continue;
+				}
+				GradleProject rootProject = build.withConnection(connection -> connection.getModel(GradleProject.class), monitor);
+				if (hasAndroidTasks(rootProject)) {
+					build.withConnection(connection -> {
+						connection.newBuild().forTasks(ASSEMBLE_TASK_NAME).run();
+						return null;
+					}, monitor);
+				}
+			} catch (Exception e) {
+				JavaLanguageServerPlugin.logException(e);
+			}
+		}
+		progressReport.setComplete(true);
+		client.sendProgressReport(progressReport);
+
+	}
+
+	private boolean hasAndroidTasks(GradleProject project) {
+		for (GradleTask task : project.getTasks()) {
+			if ("Android".equals(task.getGroup())) {
```

Comment:
will `equalsIgnoreCase` safer?

Reply:
Not found any reference about task group in Android official API. But for other task groups, they can be either in uppercase or lowercase, so let's use `equalsIgnoreCase` to avoid possible former versions using "android"